### ISSUE TITLE
Avoid special casing work items in `sendtohelixhelp.proj`

### ISSUE THREAD

**Issue by radical** [dotnet/runtime#46156]
```xml
    <ItemGroup Condition="'$(TargetOS)' == 'Browser'">
      <!-- Create a work item for run-only WASM apps  -->
      <_RunOnlyWorkItem Include="$(TestArchiveRoot)runonly/**/*.zip" />
      <HelixWorkItem Include="@(_RunOnlyWorkItem -> '%(FileName)')" >
        <PayloadArchive>%(Identity)</PayloadArchive>
        <!-- No RunTests script generated for the sample project so we just use the direct command -->
        <Command>dotnet exec $XHARNESS_CLI_PATH wasm $XHARNESS_COMMAND  --app=. --engine=V8  --engine-arg=--stack-trace-limit=1000 --js-file=runtime.js --output-directory=$XHARNESS_OUT -- --run WasmSample.dll</Command>
      </HelixWorkItem>
    </ItemGroup>
```

Instead of this, we could try generating `RunTests.sh` which is done by https://github.com/dotnet/runtime/blob/93e767bd18975bf33d64177f6849c12d7505d206/eng/testing/tests.targets#L51-L52 .

Based on @safern's comment https://github.com/dotnet/runtime/pull/45768#issuecomment-746758470
```
I think I would rather do that and try to be consistent, it seems cumbersome to have that "workaround" in sendhelix to include the sample as part of the payload and hardcode the HelixCommand.

Can we open a follow up issue to clean this up and make it more generic for more samples? I see we also added this for iOS and it is doing somewhat similar instead of use the targets we already have.
```

**Comment by premun** [comment]
@radical please note that `RunTest.sh` is not used for Helix but only in local development.

For iOS and Android, we use the Helix SDK to take care of running XHarness.

For iOS, we have something like `RunTest.sh` (but static):
https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/ios-helix-job-payload.sh

For Android, we call XHarness directly in the Helix command:
https://github.com/dotnet/arcade/blob/16b1ec7dc37cd19927eb7a25e76a67236a9c74a0/src/Microsoft.DotNet.Helix/Sdk/CreateXHarnessAndroidWorkItems.cs#L106
but to my best knowledge, @MattGal might be adding a similar script soon to do some emulator restarting.

For WASM, I would suggest to go similar route and add support on Arcade level as well.

**Comment by safern** [comment]
@premun yeah that's correct. However for iOS we should try to also binplace the sample app into `TestArchiveTestsRoot` so that we don't have to duplicate logic in `sendtohelixhelp.proj`: https://github.com/dotnet/runtime/blob/03e4f5e36410203a7ebab55ac177c166406ea1fe/src/libraries/sendtohelixhelp.proj#L192

**Comment by MaximLipnin** [comment]
@safern As for iOS, could you point me to other way besides putting to separate directories to differentiate the sample from test apps in order to set a couple of specific properties?

**Comment by mdh1418** [comment]
@radical I took a look and it seems like there were significant modifications to sendtohelix*(help/mobile/wasm).proj since this issue was first created. I'm not sure if I'm interpreting things correctly. It seems like this issue aims to remove hardcoded/special cased `HelixWorkItem` by using `RunTest.sh` generated through `GenerateRunScript`, but comments suggest this is more of an effort to remove duplicate code? 

Is this still relevant, might be worth pushing to 8 if it isn't urgent.

**Comment by radical** [comment]
@mdh1418 I think we have reduced that quite a bit. But @premun's suggestion in https://github.com/dotnet/runtime/issues/46156#issuecomment-746816456, we should probably still do that. But it's not urgent at all. Moving to Future milestone.
